File: app/src/main/java/com/example/hwsw_lab2/UserHomepage.java
package com.example.hwsw_lab2;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;
import android.view.MenuItem;

import com.example.hwsw_lab2.fragment.HomepageFragment;
import com.example.hwsw_lab2.fragment.ProfileFragment;
import com.example.hwsw_lab2.fragment.Sleep_status;
import com.example.hwsw_lab2.fragment.SurveyFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class UserHomepage extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_homepage);

        BottomNavigationView bottomNav = findViewById(R.id.bottom_navigation);
        bottomNav.setOnNavigationItemSelectedListener(navListener);

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
                    new HomepageFragment()).commit();
        }
//
//        HomepageFragment homepageFragment = new HomepageFragment();
//        FragmentManager fm = getSupportFragmentManager();
//
//        fm.beginTransaction().add(R.id.user_homepage_layout, homepageFragment).commit();


    }

    private BottomNavigationView.OnNavigationItemSelectedListener navListener = new BottomNavigationView.OnNavigationItemSelectedListener() {
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            Fragment selectedFragment = null;

            switch (item.getItemId()){
                case R.id.ic_home:
                    selectedFragment = new HomepageFragment();
                    break;

                case R.id.ic_profile:
                    selectedFragment = new ProfileFragment();
                    break;

                case R.id.ic_black_panther:
                    selectedFragment = new SurveyFragment();
                    break;

            }

            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
                    selectedFragment).commit();
            return true;
        }
    };
}

File: app/src/main/java/com/example/hwsw_lab2/UserHomepage.kt
package com.example.hwsw_lab2

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.example.hwsw_lab2.fragment.HomepageFragment
import com.example.hwsw_lab2.fragment.ProfileFragment
import com.example.hwsw_lab2.fragment.SurveyFragment
import kotlinx.android.synthetic.main.activity_user_homepage.*

class UserHomepage : AppCompatActivity() {

    private val homepageFragment = HomepageFragment()
    private val profileFragment = ProfileFragment()
    private val surveyFragment = SurveyFragment()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user_homepage)
        replaceFragment(homepageFragment)

        bottom_navigation.setOnNavigationItemReselectedListener {
            when(it.itemId){
                R.id.ic_home -> replaceFragment(homepageFragment)
                R.id.ic_profile -> replaceFragment(profileFragment)
                R.id.ic_survey -> replaceFragment(surveyFragment)
            }
            true
        }
    }

    private fun replaceFragment(fragment: Fragment){
        if (fragment != null){
            val transaction = supportFragmentManager.beginTransaction()
            transaction.replace(R.id.fragment_container, fragment)
            transaction.commit()
        }
    }

}